www_src/components/basic-element/basic-element.jsx

```diff
@@ -11,13 +11,29 @@ var touchhandler = require("../../lib/touchhandler");
 var dispatcher = require('../../lib/dispatcher');
 
 var BasicElement = React.createClass({
-
   statics: {
     types: {
       image: require('./types/image.jsx'),
       text: require('./types/text.jsx'),
       link: require('./types/link.jsx')
-    }
+    },
+    /**
+     * A static function for determining whether a given scale will
+     * be safe, or needs capping because it would lead to an element
+     * that is visually too small to manipulate.
+     */
+    safifyScale: (function() {
+      var minScaledEdgeLength = 44;
+      return function(component, scale) {
+        var node  = component.getDOMNode(),
+            style = getComputedStyle(node,null),
+            w = parseInt(style.getPropertyValue("width"), 10),
+            h = parseInt(style.getPropertyValue("height"), 10),
+            e = w>h ? w : h;
+        if(e*scale < minScaledEdgeLength) { scale = minScaledEdgeLength/e; }
+        return scale;
+      };
+    }())
```

Comment:
closure around minScaledEdgeLength  - if we use a static, that is no longer necessary